infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/CorfuReplicationClusterManagerBaseAdapter.java

```diff
@@ -0,0 +1,52 @@
+package org.corfudb.infrastructure.logreplication.infrastructure.plugins;
+
+import lombok.Getter;
+
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.infrastructure.logreplication.infrastructure.CorfuReplicationDiscoveryServiceAdapter;
+import org.corfudb.infrastructure.logreplication.proto.LogReplicationClusterInfo.TopologyConfigurationMsg;
+
+/***
+ * This is the base class for CorfuReplicationSiteManager and implements the basic functionality.
+ * Any SiteMangerImplementation should extend this class or implements the interface.
+ *
+ */
+@Slf4j
+public abstract class CorfuReplicationClusterManagerBaseAdapter implements CorfuReplicationClusterManagerAdapter {
+    @Getter
+    CorfuReplicationDiscoveryServiceAdapter corfuReplicationDiscoveryService;
+
+    @Getter
+    TopologyConfigurationMsg topologyConfig;
+
+    String localEndpoint;
+
+    public void start(CorfuReplicationDiscoveryServiceAdapter corfuReplicationDiscoveryService) {
+        this.corfuReplicationDiscoveryService = corfuReplicationDiscoveryService;
+        register();
```

Comment:
does this register the discovery service?  If so, my suggestion is to invoke it from discovery service instead.